stm32-modules/include/flex-stacker/flex-stacker/motor_task.hpp

```diff
@@ -299,25 +295,8 @@ class MotorTask {
         static_cast<void>(policy);
         // sent from the driver task so we know we've written to driver
         // successfully
-        switch (m.motor_id) {
-            case MotorID::MOTOR_X:
-                _x_mech_conf.microstep =
-                    static_cast<float>(pow(2, m.microsteps_power));
-                _x_state.steps_per_mm = _x_mech_conf.get_usteps_per_mm();
-                break;
-            case MotorID::MOTOR_Z:
-                _z_mech_conf.microstep =
-                    static_cast<float>(pow(2, m.microsteps_power));
-                _z_state.steps_per_mm = _z_mech_conf.get_usteps_per_mm();
-                break;
-            case MotorID::MOTOR_L:
-                _l_mech_conf.microstep =
-                    static_cast<float>(pow(2, m.microsteps_power));
-                _l_state.steps_per_mm = _l_mech_conf.get_usteps_per_mm();
-                break;
-            default:
-                break;
-        }
+        motor_state(m.motor_id).lms_config.microstep =
+            static_cast<float>(pow(2, m.microsteps_power));
```

Comment:
I think that if you make this `pow(2.f, m.microsteps_power)` it'll remain a float without the static cast